src/containerapp/azext_containerapp/_clients.py

```diff
@@ -415,6 +415,22 @@ def get_auth_token(cls, cmd, resource_group_name, name):
         r = send_raw_request(cmd.cli_ctx, "POST", request_url)
         return r.json()
 
+    @classmethod
+    def validate_domain(cls, cmd, resource_group_name, name, hostname):
+        management_hostname = cmd.cli_ctx.cloud.endpoints.resource_manager
+        sub_id = get_subscription_id(cmd.cli_ctx)
+        url_fmt = "{}/subscriptions/{}/resourceGroups/{}/providers/Microsoft.App/containerApps/{}/listCustomHostNameAnalysis?api-version={}&customHostname={}"
+        request_url = url_fmt.format(
+            management_hostname.strip('/'),
+            sub_id,
+            resource_group_name,
+            name,
+            STABLE_API_VERSION,
+            hostname)
+
+        r = send_raw_request(cmd.cli_ctx, "POST", request_url)
+        return r.json()
```

Comment:
Why are these requests not considered to be implemented in the Python SDK method, but directly request the REST service

Reply:
@zhoxing-ms our plan with the `az containerapp` commands was to use raw requests while in the extension, and then use the SDK when we put this in a core CLI command module -- similar to the `appservice-kube` commands. This has helped us keep the extension install size down and add features independently of their support in the SDK. We were under the impression that using raw requests is not allowed in the core CLI but permissible in CLI extensions

Reply:
@zhoxing-ms , @StrawnSC is right - the swagger & Python SDK changes were too delayed for us to take a dependency on this. POST GA we plan to move this to using SDK completely + maybe move to core-cli as well

Reply:
Got it, thanks~